stl/inc/mutex

```diff
@@ -299,22 +299,56 @@ void _Unlock_locks(const int _First, const int _Last, index_sequence<_Indices...
     (void) _Ignored;
 }
 
+template <class _Fn>
+struct _Unlock_functor_guard {

```

Comment:
I do not love this name. I'd prefer something along the lines of `_Call_guard`, since that's what it does - it calls the function object passed in.